File: backend/blackbox/src/main/java/com/insight_web/blackbox/usecase/SiteSelection.java
package com.insight_web.blackbox.usecase;

import com.insight_web.blackbox.domain.Site;
import com.insight_web.blackbox.usecase.storage.SiteStorage;
import com.insight_web.blackbox.usecase.storage.UserStorage;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class SiteSelection {
    private final UserStorage userStorage;
    private final SiteStorage siteStorage;
    public Site select(String siteId) {
        return siteStorage.findById(siteId);
    }

    public List<Site> selectAllByOwnerId(String ownerEmail) {
        var ownerId = userStorage.findByEmail(ownerEmail).getId();
        return siteStorage.findAllByOwnerId(ownerId);
    }
}

File: backend/auth/src/main/kotlin/com/insightweb/auth/usecase/SiteSelection.kt
package com.insightweb.auth.usecase

import com.insightweb.auth.domain.Site
import com.insightweb.auth.usecase.storage.SiteStorage
import com.insightweb.auth.usecase.storage.UserStorage
import jakarta.transaction.Transactional
import org.springframework.stereotype.Service

@Service
class SiteSelection(
    private val siteStorage: SiteStorage,
    private val userStorage: UserStorage,
) {

    fun select(url: String): Site = siteStorage.findById(url)
    @Transactional
    fun selectAllNyOwner(ownerEmail: String): List<Site> {
        val id = userStorage.findByEmail(ownerEmail).id
        return siteStorage.findAllByOwnerId(id)
    }
}